bridge/checkpoint/checkpoint.go

```diff
@@ -0,0 +1,138 @@
+package checkpoint
+
+import (
+	"github.com/0xPolygon/polygon-edge/bridge/checkpoint/transport"
+	"github.com/0xPolygon/polygon-edge/bridge/sam"
+	"github.com/0xPolygon/polygon-edge/network"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+)
+
+const (
+	EpochSize = 100
+)
+
+type Checkpoint interface {
+	Start() error
+	Close() error
+	CreateNewCheckpoint() error
+	SetValidators([]types.Address)
+}
+
+type Blockchain interface {
+	GetBlocks(start, end uint64) ([]types.Block, error)
+}
+
+type checkpoint struct {
+	logger            hclog.Logger
+	signer            sam.Signer
+	blockchain        Blockchain
+	rootchainContract RootChainContractClient
+	transport         transport.CheckpointTransport
+	validators        []types.Address
+
+	closeCh chan struct{}
+}
+
+func NewCheckpoint(
+	logger hclog.Logger,
+	network *network.Server,
+	signer sam.Signer,
+) (Checkpoint, error) {
+	checkpointLogger := logger.Named("checkpoint")
+
+	return &checkpoint{
+		logger: checkpointLogger,
+		signer: signer,
+		// blockchain: blockchain,
+		// rootchainContract: rootchainContractClient,
+		transport: transport.NewLibp2pGossipTransport(logger, network),
+	}, nil
+}
+
+func (c *checkpoint) Start() error {
+	return nil
+}
+
+func (c *checkpoint) Close() error {
+	return nil
+}
+
+func (c *checkpoint) SetValidators(vals []types.Address) {
+	c.validators = vals
+}
+
+func (c *checkpoint) CreateNewCheckpoint() error {
+	// Step1: Get Latest information stored in RootChain contract
+	currentEpoch, err := c.rootchainContract.GetCurrentHeaderBlock()
+	if err != nil {
+		return err
+	}
+
+	lastChildBlock, err := c.rootchainContract.GetLastChildBlock()
```

Comment:
If we know what is currentEpoch in the root chain, do we need the last-child block? 
We can use currentEpoch*epochSize + 1 for lower bound

Reply:
@Kourin1996 Ping

Reply:
@Aleksao998 I created mock

Reply:
Thank you for an explanation!